azure-pipelines.yml

```diff
@@ -61,6 +61,44 @@ jobs:
     inputs:
       testResultsFiles: '**/test-*.xml'
       testRunTitle: 'Publish test results for Python $(python.version)'
+- job: static_analysis
+  pool:
+    vmIMage: macOS-10.13
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '3.7'
+      addToPath: true
+      architecture: 'x64'
+  - script: /bin/bash -c "sudo xcode-select -s /Applications/Xcode_10.app/Contents/Developer"
+    displayName: 'select Xcode version'
+  - script: python -m pip install --upgrade pip setuptools wheel cython
+    displayName: 'Install tools / dependencies'
+  # we build NumPy for static analysis because of the template *.src files
+  - script: python setup.py build -j 4 install
+    displayName: 'Build NumPy'
+    env:
+      BLAS: None
+      LAPACK: None
+      ATLAS: None
+      ACCELERATE: None
+      CC: /usr/bin/clang
+  - script: brew install cppcheck
+    displayName: 'Install cppcheck static analyzer'
+  - script: time cppcheck --enable=all . >& cppcheck_results.txt
+    displayName: 'run cppcheck static analyzer'
+  # start by checking for bugs flagged by cppcheck with (error), but
+  # ignore the large number of "order of evaluation of side effects" issues
+  # detected in lapack_lite (NOTE: are those real issues?)
```

Comment:
Can you give an example of this? It's very plausible that `f2c` produces bad output, and also very plausible that the original fortran (which is a fairly old version of lapack) has the same problem.

Reply:
`[numpy/linalg/lapack_lite/f2c_c_lapack.c:24393]: (error) Expression 'r__1=d__[finish],(double)((r__1)>=0?r__1:-(r__1))' depends on order of evaluation of side effects `

`(error)` should mean bug

The full raw `cppcheck` output was linked in the original PR message